src/main/java/teammates/ui/template/FeedbackResultsTable.java

```diff
@@ -26,6 +28,42 @@ public FeedbackResultsTable(int fbIndex, String studentName, FeedbackSessionResu
                 this.receivedResponses.add(new FeedbackResponsePersonRow(fbIndex, giverIndex, entry.getKey(), "giver",
                                                                          entry.getValue(), result));
             }
+
+            boolean[] hasAnsweredContrib = { false };
+            boolean[] contribExistsFromOtherStudent = { false };
+            FeedbackResponseAttributes[] response = new FeedbackResponseAttributes[1];
+
+            received.forEach((key, value) -> {
+                for (FeedbackResponseAttributes feedbackResponseAttributes : value) {
+                    FeedbackQuestionType questionType = feedbackResponseAttributes.feedbackQuestionType;
+                    String recipient = feedbackResponseAttributes.recipient;
+                    String giver = feedbackResponseAttributes.giver;
+                    if (questionType.equals(FeedbackQuestionType.CONTRIB)) {
+                        response[0] = new FeedbackResponseAttributes(feedbackResponseAttributes);
+                        contribExistsFromOtherStudent[0] = true;
+                    }
+                    if (giver.equals(recipient) && questionType.equals(FeedbackQuestionType.CONTRIB)) {
+                        hasAnsweredContrib[0] = true;
+                    }
+                }
+            });
+            // if student has not submitted the session or CONTRIB response
+            // about himself then student's submission to CONTRIB is shown No
+            // Response and PC is shown next to it
+            if ((!hasStudentSubmitted || !hasAnsweredContrib[0]) && contribExistsFromOtherStudent[0]) {
+                giverIndex++;
+                response[0].giver = response[0].recipient;
+                response[0].giverSection = response[0].recipientSection;
+                List<FeedbackResponseAttributes> responses = new ArrayList<>();
+                responses.add(response[0]);
+                FeedbackResponsePersonRow contribPcRow = new FeedbackResponsePersonRow(fbIndex, giverIndex, studentName,
+                                                                                        "giver", responses, result);
+                String responseText = contribPcRow.getResponses().get(0).getResponseText();
```

Comment:
I'm still not convinced. Which condition states that `contribPcRow.getResponses()` will not be empty?
If this is the same as the `responses` variable just few lines before this, then will it not be better to use `responses` instead of `contribPcRow.getResponses()`?